README.md

```diff
@@ -19,6 +19,113 @@ NumPyro is designed to be *lightweight* and focuses on providing a flexible subs
+
+```python
+>>> nuts_kernel = NUTS(eight_schools)
+>>> mcmc = MCMC(nuts_kernel, num_warmup=500, num_samples=1000)
+>>> rng_key = random.PRNGKey(0)
+>>> mcmc.run(rng_key, J, sigma, y=y, extra_fields=('diverging',))
+```
+
+We can print the summary of the MCMC run, and examine if we observed any divergences during inference:
+
+```python
+mcmc.print_summary()
+
+                mean       std    median      5.0%     95.0%     n_eff     r_hat
+        mu      3.94      2.81      3.16      0.03      9.28    114.51      1.06
+       tau      3.20      2.97      2.40      0.38      7.28     24.06      1.07
+  theta[0]      5.56      5.26      4.10     -1.67     13.52     63.57      1.05
+  theta[1]      4.48      4.15      3.26     -2.44     11.25    148.63      1.05
+  theta[2]      3.62      4.40      3.26     -3.85     10.75    445.91      1.01
+  theta[3]      4.25      4.24      3.24     -2.99     10.68    366.29      1.04
+  theta[4]      3.25      3.94      3.29     -3.34      9.84    311.03      1.00
+  theta[5]      3.66      4.27      2.77     -2.79     11.06    344.57      1.02
+  theta[6]      5.74      4.67      4.34     -1.92     13.25     58.42      1.05
+  theta[7]      4.29      4.63      3.23     -2.14     12.37    342.50      1.02
+
+>>> print("Number of divergences: {}".format(sum(mcmc.get_extra_fields()['diverging'])))
+
+Number of divergences: 139
+```
+
+The values above 1 for the split Gelman Rubin diagnostic (`r_hat`) indicates that the chain has not fully converged. The low value for the effective sample size (`n_eff`), particularly for `tau`, and the number of divergent transitions looks problematic. Fortunately, this is a common pathology that can be rectified by using a [non-centered paramaterization](https://mc-stan.org/docs/2_18/stan-users-guide/reparameterization-section.html) for `tau` in our model. This is straightforward to do in NumPyro by using a [TransformedDistribution](http://num.pyro.ai/en/latest/distributions.html#transformeddistribution) instance. Let us rewrite the same model but instead of sampling `theta` from a `Normal(mu, tau)`, we will instead sample it from a base `Normal(0, 1)` distribution that is transformed using an [AffineTransform](http://num.pyro.ai/en/latest/distributions.html#affinetransform). Note that by doing so, NumPyro runs HMC by generating samples for the base `Normal(0, 1)` distribution instead. We see that the resulting chain does not suffer from the same pathology — the Gelman Rubin diagnostic is 1 for all the parameters and the effective sample size looks quite good! 
+
+```python
+>>> # Eight Schools example - Non-centered Reparametrization
+... def eight_schools_noncentered(J, sigma, y=None):
+...     mu = numpyro.sample('mu', dist.Normal(0, 5))
+...     tau = numpyro.sample('tau', dist.HalfCauchy(5))
+...     with numpyro.plate('J', J):
+...         theta = numpyro.sample('theta', 
+...                                dist.TransformedDistribution(dist.Normal(0., 1.),
```

Comment:
Could you confirm if we get the same result with `dist.Normal(np.zeros(J), 1.)`? :D I am a bit curious about it.

Reply:
We get similar results (good), but it is not exactly the same. I don't know off the top of my head why that's the case though - it seems to me that we should draw the same sequence of random numbers in either case.

Reply:
Think about this again, I guess it comes from [init_strategy](https://github.com/pyro-ppl/numpyro/blob/master/numpyro/infer/util.py#L202). Let me go deeper into this issue to make the behavior consistent.

Btw, I want to confirm that hmm/covtype benchmark is still good (indeed we get better results) after the change at the terminate condition. :D But the flowhmc work is 5x slower (trees are bigger). Do you think that it is a blocker that we need to look into before the release?

Reply:
> But the flowhmc work is 5x slower (trees are bigger). Do you think that it is a blocker that we need to look into before the release?

I think flowhmc is mostly for internal research, so I'm not too concerned about that, but we should look into why this results in 5X larger trees. We can release a patch shortly after if you think there are other kinds of models that might be affected.

Reply:
I can confirm that the difference is due to precision. The following are the values of 4th element of `theta` in two versions. We can see that after 5 steps, the values in two versions are a bit different.
```
[(-2.2974488735198975, -2.2974488735198975),
 (-2.2974488735198975, -2.2974488735198975),
 (-2.320910930633545, -2.320910930633545),
 (-1.7636312246322632, -1.7636312246322632),
 (3.898336887359619, 3.898348093032837),
 (-2.118842363357544, -2.118828773498535),
 (-12.201387405395508, -12.20230484008789),
 (-9.108250617980957, -9.103801727294922),
 (-9.108250617980957, -9.103801727294922),
 (7.62028694152832, 7.620795249938965),
```

Reply:
> Probably, (x - 0) / 1 will not be equal to (x - np.zeros(8)) / np.ones(8), or how gradients are numerically accumulated...

Hmm..yeah I think the code generated for the backward pass may be different or the sequencing of broadcasting operations is different. The code generated by the tracer for the two versions is not identical for some reason.